### PR TITLE
[Merged by Bors] - chore: split order synonym instances for GroupWithZero into their own file

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
This is consistent with the `Group` file, and drops the dependency on `Mathlib.Order` entirely.

---

- [x] depends on: #13008 
- [x] depends on: #13033 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/GroupWithZero/OrderSynonym.lean
Comment (YaelDillies) [inline Mathlib/Algebra/GroupWithZero/OrderSynonym.lean:1]: Can you rename that to `Algebra.Order.GroupWithZero.Synonym`? Nothing order theoretic should be under `Algebra` without being under `Algebra.Order`

**Review by YaelDillies — COMMENTED** [review]
Can you add some `assert_not_exists` proving that you have done something?

**Comment by Ruben-VandeVelde** [comment]
Will do.

File: Mathlib/Algebra/GroupWithZero/Basic.lean [hunk 1]
```diff
@@ -36,6 +35,8 @@ and require `0⁻¹ = 0`.
 
 -/
 
+assert_not_exists OrderDual
+assert_not_exists Lex
```
Comment (YaelDillies) [inline Mathlib/Algebra/GroupWithZero/Basic.lean:39]: Does this stronger one hold?
```suggestion
assert_not_exists DenselyOrdered
```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by Ruben-VandeVelde** [comment]
maintainer merge

(from Yaël earlier; just needed to merge master)

**Comment by kbuzzard** [comment]
Thanks a lot for this busywork!

bors merge

**Pull request closed without merging**